mergify_engine/context.py

```diff
@@ -82,6 +82,65 @@ def get_repository(self, name: github_types.GitHubRepositoryName) -> "Repository
             self.repositories[name] = repository
         return self.repositories[name]
 
+    TEAM_MEMBERS_CACHE_KEY_PREFIX = "team_members"
+    TEAM_MEMBERS_CACHE_KEY_DELIMITER = "/"
+    TEAM_MEMBERS_EXPIRATION = 3600  # 1 hour
+
+    @classmethod
+    def _team_members_cache_key_for_repo(
+        cls,
+        owner_id: github_types.GitHubAccountIdType,
+        team_slug: github_types.GitHubTeamSlug,
+    ) -> str:
+        return (
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_PREFIX}"
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_DELIMITER}{owner_id}"
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_DELIMITER}{team_slug}"
+        )
+
+    @classmethod
+    async def clear_team_members_cache_for_team(
+        cls,
+        redis: utils.RedisCache,
+        owner: github_types.GitHubAccount,
+        team_slug: github_types.GitHubTeamSlug,
+    ) -> None:
+        await redis.delete(
+            cls._team_members_cache_key_for_repo(owner["id"], team_slug),
+        )
+
+    @classmethod
+    async def clear_team_members_cache_for_org(
+        cls, redis: utils.RedisCache, user: github_types.GitHubAccount
+    ) -> None:
+        pipeline = await redis.pipeline()
+        async for key in redis.scan_iter(
+            f"{cls.TEAM_MEMBERS_CACHE_KEY_PREFIX}{cls.TEAM_MEMBERS_CACHE_KEY_DELIMITER}{user['id']}{cls.TEAM_MEMBERS_CACHE_KEY_DELIMITER}*"
+        ):
+            await pipeline.delete(key)
+        await pipeline.execute()
+
+    async def get_team_members(
+        self, team_slug: github_types.GitHubTeamSlug
+    ) -> typing.List[github_types.GitHubLogin]:
+
+        key = self._team_members_cache_key_for_repo(self.owner_id, team_slug)
+        members: typing.List[github_types.GitHubLogin] = await self.redis.lrange(
+            key, 0, -1
+        )
+        if not members:
+            members = [
+                github_types.GitHubLogin(member["login"])
+                async for member in self.client.items(
+                    f"/orgs/{self.owner_login}/teams/{team_slug}/members"
+                )
+            ]
+            pipe = await self.redis.pipeline()
+            await pipe.rpush(key, *members)
+            await pipe.expire(key, self.TEAM_MEMBERS_EXPIRATION)
```

Comment:
I feel like this is going to crash if the team is empty and has 0 member.

Reply:
I have added a test to covert that

Reply:
push it!